sdv/data_processing/data_processor.py

```diff
@@ -631,9 +630,14 @@ def fit(self, data):
             data (pandas.DataFrame):
                 Table to be analyzed.
         """
+        if data.empty:
+            raise ValueError('The fit dataframe is empty, transformer will not be fitted.')
```

Comment:
minor: can we say synthesizer will not be fitted, since that's the path the user will take to end up seeing this message?